autogpt_platform/frontend/src/components/agptui/LibrarySearchBar.tsx

```diff
@@ -0,0 +1,101 @@
+"use client";
+import { Search, X } from "lucide-react";
+import { Input } from "../ui/input";
+import { useRef, useState } from "react";
+import { motion, AnimatePresence } from "framer-motion";
+import debounce from "lodash/debounce";
+import { useBackendAPI } from "@/lib/autogpt-server-api/context";
+import { useLibraryPageContext } from "./providers/LibraryAgentProvider";
+
+export const LibrarySearchBar = () => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [isFocused, setIsFocused] = useState(false);
+  const api = useBackendAPI();
+  const { setAgentLoading, setAgents, libraryFilter, setSearchTerm } =
+    useLibraryPageContext();
+
+  const debouncedSearch = debounce(async (value: string) => {
+    try {
+      setAgentLoading(true);
+      setSearchTerm(value);
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+      const response = await api.listLibraryAgents({
+        search_term: value,
+        sort_by: libraryFilter,
+        page: 1,
+      });
+      setAgents(response.agents);
+      setAgentLoading(false);
+    } catch (error) {
+      console.error("Search failed:", error);
+    }
+  }, 300);
+  const handleSearchInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const searchTerm = e.target.value;
+    debouncedSearch(searchTerm);
+  };
+
+  return (
```

Comment:
is this fundamentally different than the other search bar? why have two? can we push back on design to keep a subset of components and move faster

Reply:
I've moved all the library specific components into `components/library`. We can look into unifying/generalizing similar implementations of the same thing later, e.g. search bars.

Reply:
- #9515